fenix/app/src/main/java/org/mozilla/fenix/compose/TabCounter.kt

```diff
@@ -0,0 +1,121 @@
+
+private const val MAX_VISIBLE_TABS = 99
+private const val SO_MANY_TABS_OPEN = "∞"
+private val NORMAL_TABS_BOTTOM_PADDING = 0.5.dp
+private const val ONE_DIGIT_SIZE_RATIO = 0.5f
+private const val TWO_DIGITS_SIZE_RATIO = 0.4f
+private const val MIN_SINGLE_DIGIT = 0
+private const val MAX_SINGLE_DIGIT = 9
+private const val TWO_DIGIT_THRESHOLD = 10
+private const val TAB_TEXT_BOTTOM_PADDING_RATIO = 4
+
+/**
+ * UI for displaying the number of opened tabs.
+ *
+ * @param tabCount the number to be displayed inside the counter.
+ */
+
+@Composable
+fun TabCounter(tabCount: Int) {
+    val normalTabCountText: String
+    val tabCountTextRatio: Float
+    val needsBottomPaddingForInfiniteTabs: Boolean
+
+    when (tabCount) {
+        in MIN_SINGLE_DIGIT..MAX_SINGLE_DIGIT -> {
+            normalTabCountText = tabCount.toString()
+            tabCountTextRatio = ONE_DIGIT_SIZE_RATIO
+            needsBottomPaddingForInfiniteTabs = false
+        }
+
+        in TWO_DIGIT_THRESHOLD..MAX_VISIBLE_TABS -> {
+            normalTabCountText = tabCount.toString()
+            tabCountTextRatio = TWO_DIGITS_SIZE_RATIO
+            needsBottomPaddingForInfiniteTabs = false
+        }
+
+        else -> {
+            normalTabCountText = SO_MANY_TABS_OPEN
+            tabCountTextRatio = ONE_DIGIT_SIZE_RATIO
+            needsBottomPaddingForInfiniteTabs = true
+        }
+    }
+
+    val normalTabsContentDescription = if (tabCount == 1) {
+        stringResource(id = R.string.mozac_tab_counter_open_tab_tray_single)
+    } else {
+        stringResource(
+            id = R.string.mozac_tab_counter_open_tab_tray_plural,
+            tabCount.toString(),
+        )
+    }
+
+    val counterBoxWidthDp =
+        dimensionResource(id = mozilla.components.ui.tabcounter.R.dimen.mozac_tab_counter_box_width_height)
+    val counterBoxWidthPx = LocalDensity.current.run { counterBoxWidthDp.roundToPx() }
+    val counterTabsTextSize = (tabCountTextRatio * counterBoxWidthPx).toInt()
+
+    val normalTabsTextModifier = if (needsBottomPaddingForInfiniteTabs) {
+        val bottomPadding = with(LocalDensity.current) { counterTabsTextSize.toDp() / TAB_TEXT_BOTTOM_PADDING_RATIO }
+        Modifier.padding(bottom = bottomPadding)
+    } else {
+        Modifier.padding(bottom = NORMAL_TABS_BOTTOM_PADDING)
+    }
+
+    Box(
+        modifier = Modifier
+            .semantics(mergeDescendants = true) {
+                testTag = TabsTrayTestTag.normalTabsCounter
+            },
+        contentAlignment = Alignment.Center,
+    ) {
+        Icon(
+            painter = painterResource(
+                id = mozilla.components.ui.tabcounter.R.drawable.mozac_ui_tabcounter_box,
+            ),
+            contentDescription = normalTabsContentDescription,
+        )
+
+        Text(
+            text = normalTabCountText,
+            modifier = normalTabsTextModifier,
+            color = LocalContentColor.current.copy(alpha = LocalContentAlpha.current),
```

Comment:
can you walk me through why this change was necessary? The removal of the color is going to break the coloring in the Tabs Tray

Reply:
I am not changing the colour, it's exactly the way it is in the original [implementation](https://searchfox.org/mozilla-mobile/source/firefox-android/fenix/app/src/main/java/org/mozilla/fenix/tabstray/TabsTrayBanner.kt#367) :)

Reply:
My bad; I totally missed the content color assignment in `TabsTrayBanner`. We're good here 👍